tests/suites/test_suite_nist_kw.data

```diff
@@ -0,0 +1,444 @@
+NIST KW self test
+mbedtls_nist_kw_self_test:
+
+NIST KW init #1 wrapping AES-128: OK
+depends_on:MBEDTLS_AES_C
+mbedtls_nist_kw_setkey:MBEDTLS_CIPHER_ID_AES:128:1:0
+
+NIST KW init #2 unwrapping AES-128: OK
+depends_on:MBEDTLS_AES_C
+mbedtls_nist_kw_setkey:MBEDTLS_CIPHER_ID_AES:128:1:0
+
+NIST KW init #3 CAMELLIA-256: bad cipher
+depends_on:MBEDTLS_CAMELLIA_C
+mbedtls_nist_kw_setkey:MBEDTLS_CIPHER_ID_CAMELLIA:256:0:MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA
```

Comment:
The error should be `MBEDTLS_ERR_CIPHER_FEATURE_UNAVAILABLE` in this case. (Or better, it should be allowed.)

Reply:
NIST 800-38F says:
>For KW and KWP, the underlying block cipher shall be approved, and the block size shall be
128 bits. Currently, the AES block cipher, with key lengths of 128, 192, or 256 bits, is the only
block cipher that fits this profile.

RFC 3349 and RFC5649 define AES key wrapping and AES key wrapping with padding.
If you would refer me to a standard defining key wrapping with Camellia, I will add support for it.

Since at the time being, it is not an unsupported feature, but an invalid cipher, `MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA` is a better error code, IMHO, same as it's done in our CCM implementation.

Reply:
[RFC 3657](https://tools.ietf.org/html/rfc3657):

> Camellia key wrapping and unwrapping are done in conformance with the AES key wrap algorithm [RFC3394]

[RFC 5794](https://tools.ietf.org/html/rfc5794)

> ARIA Key-Wrap with Padding Algorithm (AES version: RFC 5649)

You can use the same algorithm with any block cipher whose block size is 128 bits.

Reply:
>You can use the same algorithm with any block cipher whose block size is 128 bits.

It's not a matter of whether I can use the same algorithm. It's a matter of standard compliance. Thanks for referring for the KW RFCs for Camellia and Aria. I will add issues for supporting these

I assume we should be doing something similar for our CCM implementation

Reply:
Created issues #1789 and #1790